packages/controller/src/resources/monitoring/tenants/grafanaDatasources.ts

```diff
@@ -346,64 +317,30 @@ http {
             `
           location / {
             ${generalProxyConfig}
-
-            # Check https://github.com/opstrace/opstrace/issues/896 for more details.
             resolver ${dnsResolver};
-            set $backend "http://query-frontend.cortex.svc.cluster.local:80$request_uri";
+            set $backend "http://query-frontend.cortex.svc.cluster.local:80";
             proxy_pass $backend;
           }
 
-          location /api/v1/rules {
+          location ~ ^/api/v1/rules(?<urlsuffix>.*)$ {
             ${generalProxyConfig}
-
-            # Check https://github.com/opstrace/opstrace/issues/896 for more details.
-            resolver ${dnsResolver};
-            set $backend "http://ruler.cortex.svc.cluster.local/prometheus/api/v1/rules$request_uri";
-            proxy_pass $backend;
-
-          }
-
-          location /api/v1/rules/ {
-            ${generalProxyConfig}
-
-            # Check https://github.com/opstrace/opstrace/issues/896 for more details.
-
-            resolver ${dnsResolver};
-            set $backend http://ruler.cortex.svc.cluster.local/prometheus/api/v1/rules/$request_uri;
-
-          }
-
-          location /api/v1/alerts {
-            ${generalProxyConfig}
-
             resolver ${dnsResolver};
-            set $backend "http://ruler.cortex.svc.cluster.local/prometheus/api/v1/alerts$request_uri";
+            set $backend "http://ruler.cortex.svc.cluster.local/prometheus/api/v1/rules$urlsuffix";
             proxy_pass $backend;
           }
 
-          location /api/v1/alerts/ {
+          location ~ ^/api/v1/alerts(?<urlsuffix>.*)$ {
             ${generalProxyConfig}
-
             resolver ${dnsResolver};
-            set $backend http://ruler.cortex.svc.cluster.local/prometheus/api/v1/alerts/$request_uri;
+            set $backend "http://ruler.cortex.svc.cluster.local/prometheus/api/v1/alerts$urlsuffix";
```

Comment:
note: review quotation marks here, should not be necessary; didn't notice before.